tests/tests/test-xcm/test-xcm-erc20-excess-gas.ts

```diff
@@ -0,0 +1,226 @@
+          },
+          fungible: amountTransferred,
+        },
+      ],
+      beneficiary: CHARLETH_ADDRESS,
+    };
+
+    const xcmMessage = new XcmFragment(config)
+      .withdraw_asset()
+      .clear_origin()
+      .buy_execution()
+      .deposit_asset_v3(2n)
+      .as_v3();
+
+    // Mock the reception of the xcm message
+    await injectHrmpMessage(context, paraId, {
+      type: "XcmVersionedXcm",
+      payload: xcmMessage,
+    } as RawXcmMessage);
+    await context.createBlock();
+
+    // Erc20 tokens should have been received
+    expect(
+      (
+        await web3EthCall(context.web3, {
+          to: contractAddress,
+          data: ERC20_INTERFACE.encodeFunctionData("balanceOf", [CHARLETH_ADDRESS]),
+        })
+      ).result
+    ).equals(bnToHex(0, { bitLength: 256 }));
+  });
+
+  it("Incoming ERC20 transfer should succeed if setting a custom gas limit", async function () {
+    const paraId = 888;
+    const paraSovereign = sovereignAccountOfSibling(context, paraId);
+    const amountTransferred = 1_000n;
+
+    const { contractAddress, rawTx } = await createContract(
+      context,
+      "ERC20ExcessGas",
+      {
+        ...ALITH_TRANSACTION_TEMPLATE,
+        gas: 5_000_000,
+      },
+      ["MyToken", "TKN", paraSovereign, ERC20_TOTAL_SUPPLY]
+    );
+    const { result } = await context.createBlock(rawTx);
+    expectEVMResult(result.events, "Succeed");
+
+    // Get pallet indices
+    const metadata = await context.polkadotApi.rpc.state.getMetadata();
+    const balancesPalletIndex = (metadata.asLatest.toHuman().pallets as Array<any>).find(
+      (pallet) => pallet.name === "Balances"
+    ).index;
+    const erc20XcmPalletIndex = (metadata.asLatest.toHuman().pallets as Array<any>).find(
+      (pallet) => pallet.name === "Erc20XcmBridge"
+    ).index;
+
+    // Send some native tokens to the sovereign account of paraId (to pay fees)
+    const { result: result2 } = await context.createBlock(
+      createTransaction(context, {
+        ...ALITH_TRANSACTION_TEMPLATE,
+        value: 1_000_000_000_000_000_000,
+        to: paraSovereign,
+        data: "0x",
+      })
+    );
+    expectEVMResult(result2.events, "Succeed");
+
+    // Create the incoming xcm message
+    const config: XcmFragmentConfig = {
+      assets: [
+        {
+          multilocation: {
+            parents: 0,
+            interior: {
+              X1: { PalletInstance: Number(balancesPalletIndex) },
+            },
+          },
+          fungible: 1_000_000_000_000_000n,
+        },
+        {
+          multilocation: {
+            parents: 0,
+            interior: {
+              X3: [
+                {
+                  PalletInstance: erc20XcmPalletIndex,
+                },
+                {
+                  AccountKey20: {
+                    network: "Any",
+                    key: contractAddress,
+                  },
+                },
+                {
+                  GeneralIndex: 300_000n,
```

Comment:
I'm guessing this is where the gas is specified? It might help to add a comment or make this an obvious variable name (e.g. `customGasLimit`)